internal/util-logging/src/main/scala/sbt/internal/util/Terminal.scala

```diff
@@ -618,6 +618,13 @@ object Terminal {
                 if (running.get) {
                   inputStream.read match {
                     case -1 =>
+                    case -2 =>
+                      // The boot server socket returns -2 if there are no connected
+                      // thin clients. If there is also no console attached, return -1.
```

Comment:
It might be better to turn this into an enum or static variable in a Java class instead of a comment.

Reply:
https://github.com/sbt/sbt/compare/a5463139e0703e0e3f22c157acd14b0c598d7619..10c549a0b950db55eb9bf9cb90cef92c393282f8